src/_pytest/fixtures.py

```diff
@@ -581,6 +581,36 @@ def _compute_fixture_value(self, fixturedef):
                     )
                 )
                 fail(msg)
+            else:
+                if hasattr(funcitem, "callspec"):
+                    # fixture can be impacted by any parametrization of the
```

Comment:
Please extract this block to a separate function to improve legibility. I think this function can return the computed `param_index`, and this would also have the benefit of having a docstring that can be used to explain what this block of code does.